### PR TITLE
Make buildlog lens regex configurable

### LINKED ISSUE #12662: Spyglass buildlog lens - configurable regex for error matching
**What would you like to be added**: Make the regex for error matching in the Spyglass buildlog lens configurable, presumably through deck config.

**Why is this needed**: Today the regex is hardcoded into the lens code and is tailored to k8s and Golang errors. For other projects using different languages, often times the most useful errors stay hidden unless the user expands the viewer.

/area prow
/area prow/spyglass
/area prow/deck

### PR DESCRIPTION
**Pull request by Katharine**
This PR makes the highlight regex used by the buildlog lens configurable, as promised in #13030. In so doing, it fixes #12662. The config looks like this:

```yaml
- required_files:
  - build-log.txt
  lens:
    name: buildlog
    config:
      highlight_regexes:
      - timed out
      - 'ERROR:'
      - (\s|^)(FAIL|Failure \[)\b
      - (\s|^)panic\b
      - ^E\d{4} \d\d:\d\d:\d\d\.\d\d\d]
```

If `highlight_regexes` is empty or not specified, or the config is still using the old style, the existing value is used by default.

This PR also changes the type of the lens config from `interface{}` to ` json.RawMessage`, making the intent more explicit.

### CODE REVIEW

File: prow/spyglass/lenses/buildlog/lens.go [hunk 1]
```diff
@@ -116,6 +122,24 @@ type BuildLogsView struct {
 	RawGetMoreRequests map[string]string
 }
 
+func getHighlightRegex(config interface{}) *regexp.Regexp {
+	var c Config
+	if err := mapstructure.Decode(config, &c); err != nil {
```
Comment (stevekuznetsov) [inline prow/spyglass/lenses/buildlog/lens.go]: Why not `json.RawMessage` and explicit decoding?

Comment (Katharine) [reply prow/spyglass/lenses/buildlog/lens.go]: Sure, sounds good to me.

**Comment by Katharine** [comment]
/retest

🙄

**Comment by stevekuznetsov** [comment]
```

44
ERROR: While resolving toolchains for target //maintenance/aws-janitor/cmd/aws-resources-list:aws-resources-list: Target //:rbe_with_network was referenced as a platform, but does not provide PlatformInfo
```

???

/retest
@krzyzacy

**Comment by Katharine** [comment]
I've got no clue what's wrong here.

**Comment by stevekuznetsov** [comment]
:bazel_fire:

**Pull request merged**